modules/packages/LinearAlgebra.chpl

```diff
@@ -1977,6 +2001,42 @@ module Sparse {
     return A;
   }
 
+  /* Adds a diagonal to a square sparse array by adding indices to
+      its sparse domain and sets the value of that diagonal to a 
+      constant. 
```

Comment:
I think now that we have changed the function name to setDiag we can make this a "no doc" function and write documentation only for the one above. A good example of is the cholesky overloads. The overload [here](https://github.com/chapel-lang/chapel/blob/master/modules/packages/LinearAlgebra.chpl#L1074) has `pragma "no doc"` and the behavior implemented by this (compilerError) is documented on the main cholesky overload right above.

The documentation for the dense case says "Sets the value of a diagonal in a dense matrix." You can change this to "Sets the value of a diagonal in a matrix. If the matrix is sparse, indices on the diagonal will be added to its domain"